### ISSUE TITLE
Reading options in rungeneric

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco#1964]
The following code fails when executed a second time from a Python shell:

```python
import cocopp
cocopp.main(cocopp.bbob.get_all('slsqp'))
```
with
<details>

```
ValueError                                Traceback (most recent call last)
<ipython-input-5-c87f1c8eaa5f> in <module>
----> 1 cocopp.main(alg_list)

/anaconda3/lib/python3.7/site-packages/cocopp-2.3.3.18.18-py3.7.egg/cocopp/rungeneric.py in main(argv)
    258         shortoptlist = list("-" + i.rstrip(":")
    259                             for i in _split_short_opt_list(shortoptlist))
--> 260         shortoptlist.remove("-o")
    261         longoptlist = list("--" + i.rstrip("=") for i in longoptlist)
    262 

ValueError: list.remove(x): x not in list
```
</details>

Adding `print(_split_short_opt_list(shortoptlist))` in line 256 of `rungeneric` yields
```
{'o:', 'h', 'v'}
```
in the first try and
```
{'-h', '-v'}
```
in the second. From here, I can't post-process anything anymore, always getting the same error.

The processing of options seems to depend on some global state it should rather not depend on, but I don't quite understand the specifics.

**Comment by nikohansen** [comment]
I think the problem is that `shortoptlist` is a global variable which is changed while processed, but somehow not reset correctly `main` is called again?

**Comment by brockho** [comment]
I saw this already before and I am quite sure I addressed this in the currently open pull request #1963. Maybe, @nikohansen, you can have a quick look and merge it into development? All tests but the usual leak check on Mac runs through.

**Comment by nikohansen** [comment]
Works in this case, though it doesn't address the root of the problem, that the options from the previous main call are used as basis for processing the options of the next main call. The evil of using global variables...

**Comment by nikohansen** [comment]
fixed with 928f69afe8ee47d02e729e77a9cd57ff9749f755
